docs/public/understanding-odo.adoc

```diff
@@ -0,0 +1,92 @@
+= Understanding `odo`
+
+`odo` is a CLI tool for creating applications on OpenShift and Kubernetes. `odo` allows developers to concentrate on creating applications without the need to administer a cluster itself.
+Creating deployment configurations, build configurations, service routes and other OpenShift or Kubernetes elements are all automated by `odo`.
+
+Existing tools such as `oc` are more operations-focused and require a deep understanding of Kubernetes and OpenShift concepts. `odo` abstracts away complex Kubernetes and OpenShift concepts allowing developers to focus on what is most important to them: code.
+
+[id="odo-key-features"]
+== Key features
+
+`odo` is designed to be simple and concise with the following key features:
+
+* Simple syntax and design centered around concepts familiar to developers, such as projects, applications, and components.
+* Completely client based. No additional server other than OpenShift is required for deployment.
+* Official support for Node.js and Java components.
+* Detects changes to local code and deploys it to the cluster automatically, giving instant feedback to validate changes in real time.
+* Lists all the available components and services from the cluster.
+
+== Core concepts
+
+Project::
+A project is your source code, tests, and libraries organized in a separate single unit.
+Application::
+An application is a program designed for end users. An application consists of multiple microservices or components that work individually to build the entire application.
+Examples of applications: a video game, a media player, a web browser.
+Component::
+A component is a set of Kubernetes resources which host code or data. Each component can be run and deployed separately.
+Examples of components: Node.js, Perl, PHP, Python, Ruby.
+Service::
+A service is software that your component links to or depends on.
+Examples of services: MariaDB, Jenkins, MySQL.
+Devfile::
+A portable file responsible for your entire reproducable development environment.
+
+[id="odo-supported-devfiles"]
+=== Official Devfiles
+
+Devfiles describe your development environment link. link:https://odo.dev/docs/deploying-a-devfile-using-odo/[Click here for more information on Devfile.]
+
```

Comment:
for odo context, devfiles more of defines the stack required to run the application

Reply:
Hi Aditi, that part was approved by the OpenShift docs team, I'm unsure if we should update this to say "stack"

Reply:
no hard opinion here, its fine.